### PR TITLE
Add script for verifying messages in MBOX files

### PR DESCRIPTION
**Pull request by sebastian-berlin-wmse**
Bug: [T305237](https://phabricator.wikimedia.org/T305237)

### CODE REVIEW

File: python_scripts/verify_mbox.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#! /usr/bin/env python3
+"""Verify that an MBOX file contains all the messages in another MBOX file.
+
+Each missing message has its id prenited to stdout and the message is
```
Comment (lokal-profil) [inline python_scripts/verify_mbox.py]: typo "printed"

File: python_scripts/verify_mbox.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    del message1["X-GM-THRID"]
+    message2 = messages2.get(message1["Message-ID"])
+    if message2 is None:
+        # Save message if it's in the first mailbox, but not the
+        # second.
+        missing.append(message1)
+        continue
+
+    del message2["X-GM-THRID"]
+    if not str(message1).strip() == str(message2).strip():
```
Comment (lokal-profil) [inline python_scripts/verify_mbox.py]: Why not `!=`

File: python_scripts/verify_mbox.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        # Check complete messages content if the id matches. Save
+        # message if there are inconsistencies.
+        missing.append(message1)
+
+if missing:
+    print(f"Messages not in second mbox:")
```
Comment (lokal-profil) [inline python_scripts/verify_mbox.py]: No need for the f-string.

Message only mentiones the first fail-case. Maybe `"Messages missing or different not in second mbox"` instead?

File: python_scripts/verify_mbox.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    box_missing = mailbox.mbox("missing.mbox")
+    for message in missing:
+        print(message["Message-ID"])
+        box_missing.add(message)
+    box_missing.close()
```
Comment (lokal-profil) [inline python_scripts/verify_mbox.py:51]: I'd add a print to end saying something like `"Missing messages outputted to missing.mbox"` to make it clear that this output was also produced.

File: python_scripts/verify_mbox.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+#! /usr/bin/env python3
+"""Verify that an MBOX file contains all the messages in another MBOX file.
```
Comment (lokal-profil) [inline python_scripts/verify_mbox.py]: To clarify either change this line (or add a new line to the docstring) clarifying that we chack that all messages in the first file are present in the second file (i.e. the order in which the files are provided matters).

**Review by lokal-profil — CHANGES_REQUESTED** [review]
Minor changes. Otherwise good

**Comment by lokal-profil** [comment]
Since each script has different requirements it might be worth adding a note about such to the docstring e.g. mailbox >= x.y.z, tqdm >= x.y.z

File: python_scripts/verify_mbox.py (hunk 2 above)
Comment (sebastian-berlin-wmse) [reply python_scripts/verify_mbox.py]: Good question🙂

File: python_scripts/verify_mbox.py (hunk 1 above)
Comment (sebastian-berlin-wmse) [reply python_scripts/verify_mbox.py]: done

File: python_scripts/verify_mbox.py (hunk 3 above)
Comment (sebastian-berlin-wmse) [reply python_scripts/verify_mbox.py]: I can add that if it helps (sans the "not"🙂).

Really it's a semantics question about what counts as "different" as opposed to "missing". No need to go into that here though.

File: python_scripts/verify_mbox.py (hunk 4 above)
Comment (sebastian-berlin-wmse) [reply python_scripts/verify_mbox.py:51]: done

File: python_scripts/verify_mbox.py (hunk 5 above)
Comment (sebastian-berlin-wmse) [reply python_scripts/verify_mbox.py]: done

**Comment by sebastian-berlin-wmse** [comment]
Added requirement for tqdm, mailbox is standard.

**Review by lokal-profil — APPROVED** [review]

**Pull request merged**